docs/endpoint.md

```diff
@@ -5,14 +5,119 @@ The endpoint is one of the components provided by Tangle-accelerator, running on
 The encrypted message would be sent to Tangle with a streaming message channel API. The streaming message channel API would ensure the order of messages in the channel. The user who wants to fetch/send message to Tangle needs to provide `data_id`, `key` and `protocol` to identify a specific message.
 A message sent by endpoint needs to be encrypted locally, which avoids message being peeked and modified. 
 
-# How to use
+# How to build endpoint
+
+## Setup Sierra development environment 
+The endpoint uses the sierra framework as based runtime system. Developers need to set up the sierra dev environment to build endpoint as specific target.
+
+### How to build wp77xx
+
+[Leaf](https://docs.legato.io/latest/toolsLeaf.html) is a workspace manager that will download, install and configure the required software packages for a Legato development environment.
+
 ```
-$ bazel build //endpoint:wp7702
-$ bazel build //endpoint:sim
+$ wget https://downloads.sierrawireless.com/tools/leaf/leaf_latest.deb -O /tmp/leaf_latest.deb && sudo apt install /tmp/leaf_latest.deb
 ```
 
-## HTTPS Connection Support
-The endpoint uses http connection as default. The message which sent to tangle-accelerator has been encrypted. So the HTTP connection would not be unsafe. To build with https connection support, add `--define https=enable` option.
+Create a workspace
+```
+$ mkdir -p workspace
+$ cd workspace
+```
+
+Setup the wp77xx target profile
+```
+$ leaf setup legato-latest -p swi-wp77_3.3.0
+```
+
+Finally, make the wp77xx endpoint target. Be careful the tangle-accelerator directory should be located at the sub level of the workspace directory created by `leaf setup legato-latest -p swi-wp77_3.3.0`
+```
+$ git clone https://github.com/DLTcollab/tangle-accelerator.git
+$ cd tangle-accelerator
+$ make TARGET=wp77xx legato # build endpoint as wp77xx target
+$ make ENABLE_ENDPOINT_TEST=true TARGET=wp77xx legato # build endpoint as wp77xx target in test mode
+```
+
+### How to build endpoint as x86 test target
```

Comment:
`How to build endpoint application fo x86`
Or just make sure the title format is the same as the `How to build ... wp77xx`.